doc/documentation.rst

```diff
@@ -1,16 +1,29 @@
 User guide
 ===========
 
-Skrub facilitates preparing tables for machine learning. It is not a
-replacement for tools such as SQL or pandas, as these tools are much more
-versatile. Rather it gives higher-level operations for machine-learning,
-typically with `scikit-learn <http://scikit-learn.org>`_ with its
-`pipelines <https://scikit-learn.org/stable/modules/compose.html>`_.
+Skrub eases preparing tables for machine learning.
+
+Starting from rich, complex data stored in one or several dataframes, it helps
+performing the data wrangling necessary to produce a numeric array that can be fed
+to a machine-learning model. This wrangling comprises joining tables (possibly
+with inexact matches), parsing text into structured data such as dates,
+extracting numeric features, etc.
+
+For those tasks, skrub does not replace a dataframe library. Instead, it
+leverages polars or pandas to provide more high-level building blocks that are
+typically needed in a machine-learning pipeline.
+
+Crucially, the transformations implemented by skrub are *stateful*: skrub
+records the transformations that were applied to the training data and replays the
+same operations when the pipeline is applied to make predictions on unseen
```

Comment:
I think that we need to state that this is important because it avoids to make data leakage leading to false interpretation or incorrect evaluation.

Reply:
Oups it was in the next sentence.